### PR TITLE
Container

### PR DESCRIPTION
**Pull request by abhi1004**
This pr is for issue #620, Print only first 12 characters of container_id

### CODE REVIEW

File: atomicapp/nulecule/container.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def extract(self, image, source, dest, update=False):
         Utils.rm_dir(tmpdir)
 
         # Clean up dummy container
-        rm_cmd = [self.docker_cli, 'rm', '-f', container_id]
+        rm_cmd = [self.docker_cli, 'rm', '-f', container_id[:12]]
```
Comment (surajssd) [inline atomicapp/nulecule/container.py]: IMO using a CONSTANT rather than hard-coding the number would be a good thing to do.

**Comment by surajssd** [comment]
@abhi1004 you can try something like:

```python
# Clean up dummy container
rm_cmd = [self.docker_cli, 'rm', '-f', container_id]
logger.debug('Removing Docker container: %s' % ' '.join(rm_cmd))
output = subprocess.check_output(rm_cmd)
logger.debug('Removed container: {}'.format(output.rstrip()[:CONTAINER_ID_LENGTH]))
```

File: atomicapp/nulecule/container.py (hunk 1 above)
Comment (cdrage) [inline atomicapp/nulecule/container.py]: Agreed with @surajssd 

In regards to this command, is it possible to have it output via the "logger.info" command?

Comment (dustymabe) [inline atomicapp/nulecule/container.py]: > In regards to this command, is it possible to have it output via the "logger.info" command?

Actually I think that would probably not be what we want.. We want to hide some of the complexity here right? I'm about to make another comment in the generic comments that should clear up what we want to do with this PR. Please read that and give me your input.

**Comment by dustymabe** [comment]
After looking at this problem closer I think we can actually leave the logger.debug statement as it currently is. The line that we want to get rid of is the uuid that gets output not through the logger at all. So what we need to do for that is capture the output and don't let it get printed to stdout. 

To do this we can easily just convert the `subprocess.call(rm_cmd)` to `
Utils.run_cmd(rm_cmd, checkexitcode=True)` or to `subprocess.check_output(rm_cmd)`. 

This should not print anything to stdout if the command succeeds.

**Comment by dustymabe** [comment]
> Though I am skeptical about having two logger.debug for the same action.

@surajssd right. probably don't need both.

**Comment by abhi1004** [comment]
@dustymabe tried `subprocess.check_output(rm_cmd)` it works.

**Comment by cdrage** [comment]
Hey @abhi1004 I'd love to get this in :) 

Are you able to update your PR with the changes from the comments? `subprocess.check_output(rm_cmd)` and we can go ahead with the merge? Remember to update the section when it's creating the container too! :dancers:

**Comment by cdrage** [comment]
Hi @abhi1004 I've created https://github.com/projectatomic/atomicapp/pull/726 to replace this :)

**Comment by abhi1004** [comment]
@cdrage Sorry I was not available since I was busy with my exams, and thanks for the update. :+1:

**Pull request closed without merging**